XamarinCommunityToolkit/Converters/ByteArrayToImageSourceConverter.shared.cs

```diff
@@ -21,34 +21,25 @@ public class ByteArrayToImageSourceConverter : ValueConverterExtension, IValueCo
         /// <returns>An object of type ImageSource.</returns>
         public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
         {
-            if (value is null) return default(ImageSource);
+            if (value is null)
+                return default(ImageSource);
 
-            byte[] imageBytes = null;
+            if (value is byte[] imageBytes)
+                return ImageSource.FromStream(() => new MemoryStream(imageBytes));
 
-            switch (value)
-            {
-                case byte[] byteArray:
-                    imageBytes = byteArray;
-                    break;
+            throw new ArgumentException("Expected value to be of type byte[].", nameof(value));
+        }
 
-                case int[] intArray:
-                    imageBytes = new byte[intArray.Length * sizeof(int)];
-                    Buffer.BlockCopy(intArray, 0, imageBytes, 0, imageBytes.Length);
-                    break;
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            var streamImageSource = (StreamImageSource)value;
```

Comment:
I guess we should check if the value is null or StreamImageSource

If null then return null
if not is StreamImageSource throw an exception

Reply:
Should be done in the latest commit.